src/query-builder/transformer/RawSqlResultsToEntityTransformer.ts

```diff
@@ -251,6 +251,20 @@ export class RawSqlResultsToEntityTransformer {
                 // we don't mark it as has data because if we will have all nulls in our object - we don't need such object
                 hasData = true
         })
+        // Set all embedded column values to null if all their child columns are null
+        if (entity) {
+            metadata.embeddeds.forEach((embedded) => {
+                if (
+                    embedded.propertyName in entity &&
+                    typeof entity[embedded.propertyName] === "object" &&
+                    Object.values(entity[embedded.propertyName]).every(
+                        (value) => value == null,
```

Comment:
```suggestion
                        (value) => value === null || value === undefined,
```

Reply:
The unstrict check was intended, if we're getting an `undefined` value (it may not even be possible, didn't have a look at the underlying implementation) from the database it should still considered as a `null` in that use case

Reply:
In that case, I would make it explicit then.
```suggestion
                        (value) => value === null || typeof value === 'undefined',
```

Reply:
would argue that this is the use case for unstrict equality 😄 
had to rework it a bit for embedded entities in [73e4db3](https://github.com/typeorm/typeorm/pull/10289/commits/73e4db35349c6161606ac9aaff86786ac66e2ab1), added a comment to explicit it instead